packages/preferences/src/browser/views/components/preference-select-input.ts

```diff
@@ -29,13 +29,14 @@ import { escapeInvisibleChars } from '@theia/core/lib/common/strings';
 export class PreferenceSelectInputRenderer extends PreferenceLeafNodeRenderer<JSONValue, HTMLDivElement> {
 
     protected readonly selectComponent = React.createRef<SelectComponent>();
+    protected readonly selectOptions: SelectOption[] = [];
 
     protected get enumValues(): JSONValue[] {
         return this.preferenceNode.preference.data.enum!;
     }
 
-    protected get selectOptions(): SelectOption[] {
-        const items: SelectOption[] = [];
+    protected updateSelectOptions(): void {
+        this.selectOptions.splice(0);
```

Comment:
Suggestion: This feels kind of hacky to me. Maybe just make the `selectOptions` non readonly and assign to it directly in `updateSelectOptions`? It's `protected` anyway.